### PR TITLE
ShadowDOM: add editors task

### PR DESCRIPTION
**Pull request by AlexKamaev**

### CODE REVIEW

File: js/core/renderer_base.js [hunk 1]
```diff
@@ -445,7 +445,7 @@ initRender.prototype.find = function(selector) {
                 const querySelector = queryId + selector.replace(/([^\\])(,)/g, '$1, ' + queryId);
                 nodes.push.apply(nodes, domAdapter.querySelectorAll(element, querySelector));
                 setAttributeValue(element, 'id', elementId);
-            } else if(domAdapter.isDocument(element)) {
+            } else if(domAdapter.isDocument(element) || domAdapter.isDocumentFragment(element)) {
```
Comment (ksercs) [inline js/core/renderer_base.js:448]: Isn't it better to patch `domAdapter.isDocument` method than update all such places?

File: js/renovation/component_wrapper/editors/editor.ts [hunk 2]
```diff
@@ -52,6 +52,12 @@ export default class Editor extends Component {
     return props;
   }
 
+  _queryChildSelector(selector: string): HTMLElement | null {
```
Comment (ksercs) [inline js/renovation/component_wrapper/editors/editor.ts]: seem like this functionality should be implement on higher component level

File: js/ui/tag_box.js [hunk 3]
```diff
@@ -1014,7 +1015,10 @@ const TagBox = SelectBox.inherit({
     _integrateInput: function() {
         this._isInputReady.resolve();
         this.callBase();
-        this._updateTagsContainer($(`.${TEXTEDITOR_INPUT_CONTAINER_CLASS}`));
+
+        const tagsContainer = $(domAdapter.getRootNode(this.element())).find('.'.concat(TEXTEDITOR_INPUT_CONTAINER_CLASS));
```
Comment (ksercs) [inline js/ui/tag_box.js]:
```suggestion
        const tagsContainer = $(domAdapter.getRootNode(this.element()))
            .find(`.${TEXTEDITOR_INPUT_CONTAINER_CLASS}`);
```

File: testing/tests/DevExpress.ui.widgets.editors/lookup.tests.js [hunk 4]
```diff
@@ -1021,7 +1021,16 @@ QUnit.module('Lookup', {
         mouse.start().down().move(0, 10).up();
 
         openPopupWithList(firstLookup);
-        assert.equal($('.' + LIST_ITEM_SELECTED_CLASS).length, 2);
+
+        // NOTE: in ShadowDOM mode one selected item is inside ShadowDOM
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/lookup.tests.js:1025]: why so strange behavior?

File: testing/tests/DevExpress.ui.widgets.editors/lookup.tests.js [hunk 5]
```diff
@@ -1021,7 +1021,16 @@ QUnit.module('Lookup', {
...
+        // and other is in document
+        if(QUnit.isInShadowDomMode()) {
+            assert.equal(document.querySelectorAll('.' + LIST_ITEM_SELECTED_CLASS).length, 1);
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/lookup.tests.js]:
```suggestion
            assert.strictEqual(document.querySelectorAll(`.${LIST_ITEM_SELECTED_CLASS}`).length, 1);
```

and similar for near rows

File: testing/tests/DevExpress.ui.widgets.editors/textEditorParts/common.tests.js [hunk 6]
```diff
@@ -1384,7 +1384,13 @@ QUnit.module('valueChanged should receive correct event parameter', {
         this.checkEvent = (assert, type, target, key) => {
             const event = this.valueChangedHandler.getCall(0).args[0].event;
             assert.strictEqual(event.type, type, 'event type is correct');
-            assert.strictEqual(event.target, target.get(0), 'event target is correct');
+
+            // NOTE: the cached event.target is missing if the element is in shadow dom
+            // looks like a bug in a browser
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/textEditorParts/common.tests.js:1389]: Can we understand is it a browser bug or not?

File: js/core/renderer_base.js (hunk 1 above)
Comment (AlexKamaev) [reply js/core/renderer_base.js:448]: Document provides other interface than DocumentFragment, so I wouldn't merge these both conditions in one method

File: testing/tests/DevExpress.ui.widgets.editors/textEditorParts/common.tests.js (hunk 6 above)
Comment (AlexKamaev) [reply testing/tests/DevExpress.ui.widgets.editors/textEditorParts/common.tests.js:1389]: I checked this scenario without devextreme controls and the result was same. We cannot store event data if the element target is in shadow dom. This only affects the test, not production code.

File: testing/tests/DevExpress.ui.widgets.editors/lookup.tests.js (hunk 4 above)
Comment (AlexKamaev) [reply testing/tests/DevExpress.ui.widgets.editors/lookup.tests.js:1025]: Opened lookup is in overlay while overlays work in document not in shadowDom, while closed popup still is in shadow dom.

File: js/renovation/component_wrapper/editors/editor.ts (hunk 2 above)
Comment (AlexKamaev) [reply js/renovation/component_wrapper/editors/editor.ts]: move this to utils for easy testing

**Review by ksercs — APPROVED** [review]

**Review by Zedwag — APPROVED** [review]

**Pull request merged**